plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py

```diff
@@ -0,0 +1,231 @@
+           populates the list of commands to be run by comparing
+           the `want` and `have` data with the `parsers` defined
+           for the Bgp_neighbor_address_family network resource.
+        """
+        w_nbrs = want.get("neighbors", {})
+        h_nbrs = have.get("neighbors", {})
+
+        if vrf:
+            begin_vrf = len(self.commands)
+
+        for k, w_nbr in iteritems(w_nbrs):
+            begin = len(self.commands)
+            h_nbr = h_nbrs.pop(k, {})
+            want_afs = w_nbr.get("address_family", {})
+            have_afs = h_nbr.get("address_family", {})
+
+            for k, want_af in iteritems(want_afs):
+                begin_af = len(self.commands)
+                have_af = have_afs.pop(k, {})
+
+                self.compare(parsers=self.parsers, want=want_af, have=have_af)
+
+                if len(self.commands) != begin_af or (not have_af and want_af):
+                    self.commands.insert(
+                        begin_af,
+                        self._tmplt.render(want_af, "address_family", False),
+                    )
+
+            # remove remaining items in have for replaced
+            for k, have_af in iteritems(have_afs):
+                self.addcmd(have_af, "address_family", True)
+
+            if len(self.commands) != begin:
+                self.commands.insert(
+                    begin, "neighbor {0}".format(w_nbr["neighbor_address"])
+                )
+
+        if self.state in ["overridden", "deleted"]:
+            for k, h_nbr in iteritems(h_nbrs):
+                begin = len(self.commands)
+                if not w_nbrs.pop(k, {}):
+                    have_afs = h_nbr.get("address_family", {})
+                    for k, have_af in iteritems(have_afs):
+                        self.addcmd(have_af, "address_family", True)
+                if len(self.commands) != begin:
+                    self.commands.insert(
+                        begin, "neighbor {0}".format(h_nbr["neighbor_address"])
+                    )
+
+        if vrf:
+            if len(self.commands) != begin_vrf:
+                self.commands.insert(begin, "vrf {0}".format(vrf))
+        else:
+            self._vrfs_compare(want, have)
+
+    def _vrfs_compare(self, want, have):
+        wvrfs = want.get("vrfs", {})
+        hvrfs = have.get("vrfs", {})
+        for k, wvrf in iteritems(wvrfs):
+            h_vrf = hvrfs.pop(k, {})
+            self._compare(want=wvrf, have=h_vrf, vrf=k)
+        # remove remaining items in have
+        for k, h_vrf in iteritems(hvrfs):
+            self._compare(want={}, have=h_vrf, vrf=k)
+
+    def _bgp_list_to_dict(self, data):
+        if "neighbors" in data:
+            for nbr in data["neighbors"]:
+                if "address_family" in nbr:
+                    nbr["address_family"] = {
+                        (x["afi"], x.get("safi")): x
+                        for x in nbr["address_family"]
+                    }
```

Comment:
The dict comprehensions used in this function may result in sanity failures.

Reply:
The dict/list comprehensions are only invalid for Python 2.6 tests which we don't support and are skipped.